src/main/java/ch/heigvd/pro/b04/polls/ServerPoll.java

```diff
@@ -59,22 +62,37 @@ public static boolean isAvailableWithToken(
         .orElse(false);
   }
 
+  /**
+   * Creates a new unique identifier for a {@link Question}.
+   *
+   * @param repository The repository to which we will add a new Session to
+   * @return A new unique identifier
+   */
+  public static Long getNewIdentifier(QuestionRepository repository) {
+    Long identifier = repository.findAll().stream()
+        .map(Question::getIdQuestion)
+        .map(QuestionIdentifier::getIdQuestion)
+        .max(Long::compareTo)
+        .map(id -> id + 1)
+        .orElse(1L);
+    return identifier;
+  }
+
   /**
    * Add a new {@link Question} to this {@link ServerPoll} instance.
    *
    * @param newQuestion The question to be added.
    */
-  public void addQuestion(Question newQuestion) {
+  @Transactional
+  public Question newQuestion(QuestionRepository repoQ, Question newQuestion) {
```

Comment:
It's probably better to pass the question data, rather than an already constructor `Question` instance.

Reply:
(for reference, we had used two different data types rather than one for the `ServerPoll` and the `ClientPoll`. This pattern ensures that we don't "mistakenly" output some data with improper identifiers set because the type signatures matched and we forgot to set its identifier)

Reply:
Why ? We receive a Question from the client, why not to insert it directly ?

Reply:
is has been done otherway